### PR TITLE
File browser refactor and refinements

### PR DESCRIPTION
**Pull request by NPatel10**
## Summary

Added support to apply changes to `package/projects` for all the `file-browser` apis
Also worked on loading models issue when loading from file system.
Fixed unloading of scene when it is deleted

## Checklist
- [x] Pre-push checks pass `npm run check`
  - [x] Linter passing via `npm run lint`
  - [x] Unit & Integration tests passing via `npm run test:packages`
  - [x] Docker build process passing via `npm run build-client`
- [x] If this PR is still a WIP, convert to a draft 
- [x] When this PR is ready, mark it as "Ready for review"
- [ ] Changes have been manually QA'd
- [ ] Changes reviewed by at least 2 approved reviewers

## References
- [x] #5387 
- [x] #5576 
- [x] #5629

## QA Steps

1. `git checkout pr_branch_name`
2. `npm install`
3. `npm run dev-reinit`
4. `npm run dev`

List any additional steps required to QA the changes of this PR, as well as any supplemental images or videos.

## Reviewers
@HexaField @zulqarnainhanif

### CODE REVIEW

**Comment by zulqarnainhanif** [comment]
getting below error in terminal on deleting through localhost

Screenshot 2022-03-31 at 10 07 12 PM

**Review by HexaField — CHANGES_REQUESTED** [review]

File: packages/editor/src/components/assets/FileBrowserContentPanel.tsx [hunk 1]
```diff
@@ -134,30 +134,60 @@ const FileBrowserContentPanel: React.FC<FileBrowserContentPanelProps> = (props)
...
+  }
+
+  const dropItemsOnPanel = async (data: FileDataType | DnDFileType, dropOn?: FileDataType) => {
     if (isLoading) return
+
     setLoading(true)
-    if (!folder) {
-      await FileBrowserService.addNewFolder(`${selectedDirectory}New_Folder`)
+    const path = dropOn?.isFolder
+      ? selectedDirectory.split('/')[-1] == dropOn.name
```
Comment (HexaField) [inline packages/editor/src/components/assets/FileBrowserContentPanel.tsx:147]: what is [-1] doing here?

File: packages/server-core/src/media/file-browser/file-browser.test.ts [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+      fs.unlinkSync(path.join(PROJECT_PATH, fileName))
+      fs.unlinkSync(path.join(STORAGE_PATH, fileName))
+      fs.unlinkSync(path.join(PROJECT_PATH, incrementedFileName))
+      fs.unlinkSync(path.join(STORAGE_PATH, incrementedFileName))
+      fs.unlinkSync(path.join(PROJECT_PATH, incrementedFileName_2))
+      fs.unlinkSync(path.join(STORAGE_PATH, incrementedFileName_2))
+    })
+  })
+
+  it('patche file with new data', async () => {
```
Comment (HexaField) [inline packages/server-core/src/media/file-browser/file-browser.test.ts:266]: typo: `patche` should be `patch`

**Review by HexaField — CHANGES_REQUESTED** [review]

File: packages/editor/src/components/assets/FileBrowserContentPanel.tsx (hunk 1 above)
Comment (NPatel10) [reply packages/editor/src/components/assets/FileBrowserContentPanel.tsx]: Simplified the logic

**Review by HexaField — CHANGES_REQUESTED** [review]
Uploading a file to the root of a project seems to add it as both a file and a folder.

Trying to drag any asset from the file browser to a text properties field copies incorrectly by appending `undefined` at the end of the path for example: `/image.pngundefined`

**Comment by NPatel10** [comment]
@HexaField, Not able to reproduce below one, can you provide more steps?

> Uploading a file to the root of a project seems to add it as both a file and a folder.

**Comment by HexaField** [comment]
> @HexaField, Not able to reproduce below one, can you provide more steps?
> 
> > Uploading a file to the root of a project seems to add it as both a file and a folder.

1. drag a file into the file browser panel
2. once it finishes uploading, it shows as both a folder and a file with the same name

Screenshot from 2022-04-09 12-14-32

**Comment by NPatel10** [comment]
> Uploading a file to the root of a project seems to add it as both a file and a folder.

Can we create a separate issue for this bug, Because I am not able to regenerate it in my system. Maybe an OS issue?

Screenshot 2022-03-26 121703

**Review by HexaField — APPROVED** [review]

**Pull request merged**